server_admin/topics/authentication/x509.adoc

```diff
@@ -209,6 +209,61 @@ image:images/x509-directgrant-flow.png[]
 
 image:images/x509-directgrant-flow-bindings.png[]
 
+==== Client certificate lookup
+
+When user sends HTTP requests directly to {project_name} server, it's the {appserver_name} undertow subsystem, which will establish the SSL handshake and extracts client certificate. Client certificate will be then saved to the attribute `javax.servlet.request.X509Certificate` of the HTTP request as specified in the servlet specification. {project_name} X509 authenticator will be then able to lookup the certificate from this attribute.
+
+However when {project_name} listens to HTTP requests behind the loadbalancer or reverse proxy, it might be the proxy server, which extracts the client certificate and establish the mutual SSL connection. Reverse proxy usually put the authenticated client certificate to the HTTP header of the underlying request and forwards it to the backend {project_name} server. In this case, {project_name} must be able to lookup the X.509 certificate chain from the HTTP headers instead of from the attribute of HTTP request like done for Undertow.
```

Comment:
However, when the {project_name} server listens to HTTP requests behind a load balancer or reverse proxy, it may be the proxy server which extracts the client certificate and establishes the mutual SSL connection. A reverse proxy usually puts the authenticated client certificate in the HTTP header of the underlying request and forwards it to the back end {project_name} server. In this case, {project_name} must be able to look up the X.509 certificate chain from the HTTP headers instead of from the attribute of HTTP request, as is done for Undertow.